javascript/node/selenium-webdriver/lib/webdriver.js

```diff
@@ -786,11 +786,13 @@ class WebDriver {
         let start = Date.now();
         let timer = setTimeout(function() {
           timer = null;
-          reject(
-              new error.TimeoutError(
-                  (message ? `${message}\n` : '')
-                      + 'Timed out waiting for promise to resolve after '
-                      + (Date.now() - start) + 'ms'));
+          try {
+            let timeoutMessage = message ? `${typeof message === 'function' ? message() : message}\n` : '';
+            reject(new error.TimeoutError(`${timeoutMessage}Timed out waiting for promise to resolve after ${Date.now() - start}ms`));
+          }
+          catch (ex) {
+            reject(new error.TimeoutError(`${ex.message}\nTimed out waiting for promise to resolve after ${Date.now() - start}ms`));
+          }
```

Comment:
This could be extracted to a function (since it's basically repeated below)

Reply:
Agreed, have you got a preferred directory for this function to live? Or within the same file?

Reply:
@jleyba I've extracted the function out

Reply:
@jleyba anything more you can think of adding?